models/migrations/v163.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func convertWebHookTypeToVarchar(x *xorm.Engine) error {
+	type WebHook struct {
+		Type string `xorm:"VARCHAR(16) 'type'"`
+	}
+
+	type HookTask struct {
+		Typ string `xorm:"VARCHAR(16) index"`
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	if err := recreateTable(sess, new(WebHook)); err != nil {
```

Comment:
```go
// recreateTable will recreate the table using the newly provided bean definition and move all data to that new table
// WARNING: YOU MUST PROVIDE THE FULL BEAN DEFINITION
// WARNING: YOU MUST COMMIT THE SESSION AT THE END
func recreateTable(sess *xorm.Session, bean interface{}) error {
```

This will destroy all data in the webhook and hooktask tables as this stands.